server/auth.js

```diff
@@ -0,0 +1,24 @@
+var admin = require("firebase-admin");
+admin.initializeApp({
+    projectId: "daily-focus-a7423",
+});
+function authorise(req, res, next) {
+    let idToken;
+    if (req.headers.authorization && req.headers.authorization.startsWith("Bearer ")) {
+        idToken = req.headers.authorization.split("Bearer ")[1];
+    } else {
+        return res.status(401).json({ message: "Unauthorized access" });
+    }
+
+    admin
+        .auth()
+        .verifyIdToken(idToken)
+        .then((token) => {
+            const UUID = token.user_id;
+            return res.json({ UUID });
+        })
+        .catch((err) => {
+            return res.status(401).json(err);
```

Comment:
This will cause 401 response to be sent (which is ok) but afterwards the code will continue to execute, maybe it would be better to do the res.status... followed by and error throw? And have the API methods handle it in a try/catch?

Reply:
res is gone